Objects/codeobject.c

```diff
@@ -365,6 +365,7 @@ init_code(PyCodeObject *co, struct _PyCodeConstructor *con)
 
     /* hydration */
     co->co_hydra_context = con->hydra_context;
+    Py_XINCREF(co->co_hydra_context);
```

Comment:
Typically we INCREF the value before copying it into the object (see examples above).